lib/imagemounter/utils/list_developer_image.js

```diff
@@ -0,0 +1,184 @@
+import _ from 'lodash';
+import { zip, net, env, fs } from '@appium/support';
+import { join as joinPath } from 'path';
+import axios from 'axios';
+import log from '../../logger';
+const { exists, readdir, lstat, mkdir, unlink } = fs;
+/**
+ * @typedef {Object} GithubTreeObject
+ * @property {string} path
+ * @property {string} mode
+ * @property {'blob' | 'tree'} type
+ * @property {string} sha
+ * @property {number} size
+ * @property {string} url
+ */
+
+/**
+ * @typedef {Object} GithubTreeResponse
+ * @property {string} sha
+ * @property {string} url
+ * @property {GithubTreeObject[]?} tree
+ * @property {boolean?} truncated
+ * @property {string?} node_id
+ * @property {number?} size
+ * @property {string?} content
+ * @property {string?} base64
+ */
+
+const AVAILABLE_REPOS = [
+    {
+        user: 'JinjunHan',
+        name: 'iOSDeviceSupport',
+        subFolderList: ['iOSDeviceSupport']
+    }, {
+        user: 'filsv',
+        name: 'iOSDeviceSupport',
+        subFolderList: []
+    }, {
+        user: 'iGhibli',
+        name: 'iOS-DeviceSupport',
+        subFolderList: ['DeviceSupport']
+    }
+];
+const DEFAULT_IMAGE_DIR_NAME = 'iOSDeviceSupport';
+const DEVELOPER_IMAGE_FILE_NAME = 'DeveloperDiskImage.dmg';
+const DEVELOPER_IMAGE_SIGNATURE_FILE_NAME = 'DeveloperDiskImage.dmg.signature';
+
+/**
+ * Use list api to return the file list of folder.
+ * @param {string} user github user
+ * @param {string} name the name of repository
+ * @param {string[]} subFolderList subfolder list in level order
+ * @returns {GithubTreeObject[]?} file list under target folder
+ */
+async function listGithubImageList(user, name, subFolderList = []) {
+    const initialUrl = `https://api.github.com/repos/${user}/${name}/git/trees/master`;
+    const repoUrl = `https://github.com/${user}/${name}/`;
+    const fileList = [];
+    let curUrl = initialUrl;
+    /**
+     * @type {GithubTreeObject[]?}
+     */
+    let ret = null;
```

Comment:
don't think we need `null`.  if there's a reason for it, can you pls explain?

```suggestion
    /**
     * @type {GithubTreeObject[]|undefined}
     */
    let ret;
```